### PR TITLE
Record labeler service

### PR DESCRIPTION
**Pull request by ausi**
@Toflar like this?

### CODE REVIEW

File: core-bundle/src/Event/DataContainerRecordLabelEvent.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    {
+        return $this->data;
+    }
+
+    public function getLabel(): string|null
+    {
+        return $this->label;
+    }
+
+    public function setLabel(string|null $label): void
```
Comment (Toflar) [inline core-bundle/src/Event/DataContainerRecordLabelEvent.php]: Should we call `$this->stopPropagation()` here? The first one that calls `setLabel()` wins, no? 🤔

Comment (ausi) [reply core-bundle/src/Event/DataContainerRecordLabelEvent.php]: I think I would let the listeners decide that. An extension might add something to an already existing label for example?

Comment (Toflar) [reply core-bundle/src/Event/DataContainerRecordLabelEvent.php]: Okay, let's do that then - the rest looks good to me 👍

**Comment by ausi** [comment]
This PR is also a prerequisite for https://github.com/contao/contao/pull/7010

File: core-bundle/src/Event/DataContainerRecordLabelEvent.php (hunk 1 above)
Comment (Toflar) [inline core-bundle/src/Event/DataContainerRecordLabelEvent.php]: Can we make this fluent like in other places?

**Review by Toflar — CHANGES_REQUESTED** [review]

File: core-bundle/src/Event/DataContainerRecordLabelEvent.php (hunk 1 above)
Comment (ausi) [reply core-bundle/src/Event/DataContainerRecordLabelEvent.php]: Done in 727affc4cde31fbe93b24441613277653ad3d2f1

**Review by Toflar — DISMISSED** [review]
❤️

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @ausi.

**Pull request merged**